src/main/java/com/expediagroup/hiveberg/IcebergInputFormat.java

```diff
@@ -77,7 +77,11 @@
     URI location = pathAsURI(job.get(TABLE_LOCATION));
 
     // Set defaults for virtual column
-    currentSnapshotId = table.currentSnapshot().snapshotId();
+    try {
+      currentSnapshotId = table.currentSnapshot().snapshotId();
+    } catch (NullPointerException e) {
```

Comment:
Yeah that's correct - I tried a few different things and I've found I can check if the snapshot returned from `table.currentSnapshot` is null to avoid the try/catch, and call `.snapshotId` after the check